lib/promises.js

```diff
@@ -1,5 +1,7 @@
 'use strict';
 
+var Promise = require('any-promise');
```

Comment:
`any-promise` will throw an error if it cannot find a Promise implementation, this is only the case for node versions < 0.12

As such should this require be wrapped in a guard which checks the node version before attempting the require or should it just be wrapped in a try/catch regardless